notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/KafkaSetupHandler.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.core.events;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.CreateTopicsOptions;
+import org.apache.kafka.clients.admin.CreateTopicsResult;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.errors.UnknownTopicOrPartitionException;
+
+@Slf4j
+@Builder
+public class KafkaSetupHandler {
+    public static final int DEFAULT_CREATE_TOPIC_TIMEOUT_MS = 10000;
+
+    @NonNull
+    @Getter
+    private final String bootstrapServers;
+
+    @NonNull
+    @Getter
+    private final Optional<NewTopic> newTopic;
```

Comment:
Also would be nice not to leak `NewTopic` out of this class to keep Kafka `org.apache.kafka.clients.admin` usage contained here. Seems you only need the topic name?

Reply:
Getter is for unit testing purposes. For a new topic not only the name is required but number of partitions and replication factor (which are included in NewTopic). I figure out a clearer way to handle this, including my overused Optional 👍

Reply:
Getter wasn't necessary, class was refactored